python/Ganga/GPIDev/Lib/File/MassStorageFile.py

```diff
@@ -107,7 +111,36 @@ def _on_attribute__set__(self, obj_type, attrib_name):
     def __repr__(self):
         """Get the representation of the file."""
 
-        return "MassStorageFile(namePattern='%s')" % self.namePattern
+        return "%s(namePattern='%s')" % (_getName(self), self.namePattern)
+
+    def mass_line_processor(self, line):
+        lineParts = line.split(' ')
+        pattern = lineParts[1]
+        outputPath = lineParts[2]
+        name = os.path.basename(outputPath).strip('.gz')
```

Comment:
If the purpose of this line is to split out the extension of the filename then using strip can be dangerous. Consider the following example.

```python
>>> '.gg.asjkdhajkh.zz.gz'.strip('.gz')
'asjkdhajkh'
```
strip will remove any combination of the given arg characters at either end. The proper way to achieve what I think that you want is to use `os.path.splitext` as follows:

```python
>>> os.path.splitext('.gg.asjkdhajkh.zz.gz')[0]
'.gg.asjkdhajkh.zz'
```